src/natcap/invest/datastack.py

```diff
@@ -311,77 +192,217 @@ def build_datastack_archive(args, model_name, datastack_path):
-                                 possible_path, filepath)
-                    return filepath
-                except KeyError:
-                    # turn the nested key into a nice name for a folder
-                    # e.g. args['category']['data_path'] --> category_data_path
-                    folder_prefix = nested_key[6 : -2].replace('\'][\'', '_')
-                    found_filepath = _collect_filepath(possible_path,
-                                                       data_dir,
-                                                       folder_prefix)
-
-                    # Store only linux-style filepaths.
-                    relative_filepath = os.path.relpath(
-                        found_filepath, temp_workspace).replace('\\', '/')
-                    files_found[possible_path] = relative_filepath
-                    LOGGER.debug('Processed path %s to %s',
-                                 args_param, relative_filepath)
-                    return relative_filepath
-        # It's not a file or a structure to recurse through, so
-        # just return the item verbatim.
-        LOGGER.info('Using verbatim value: %s', args_param)
-        return args_param
-
-    log_format = "%(args_key)-25s %(name)-25s %(levelname)-8s %(message)s"
-    with utils.log_to_file(logfile, log_fmt=log_format) as handler:
-        new_args = {
-            'args': _recurse(args, handler),
-            'model_name': model_name,
-            'invest_version': __version__
-        }
+                    shutil.copyfile(src_path, dest_path)
+
+            LOGGER.debug(
+                f'Directory copied from {source_path} --> {target_directory}')
+            target_arg_value = _relpath(target_directory)
+            files_found[source_path] = target_arg_value
+
+        elif input_type in spatial_types:
+            # Create a directory with a readable name, something like
+            # "aoi_path_vector" or "lulc_cur_path_raster".
+            spatial_dir = os.path.join(data_dir, f'{key}_{input_type}')
+            target_arg_value = _relpath(_copy_spatial_files(
+                source_path, spatial_dir))
+            files_found[source_path] = target_arg_value
+
+        elif input_type == 'other':
+            # Note that no models currently use this to the best of my
+            # knowledge, so better to raise a NotImplementedError
+            raise NotImplementedError(
+                'The "other" ARGS_SPEC input type is not supported')
+        else:
+            LOGGER.debug(
+                f"Type {input_type} is not filesystem-based; "
+                "recording value directly")
+            # not a filesystem-based type
+            # Record the value directly
+            target_arg_value = args[key]
+        rewritten_args[key] = target_arg_value
+
+    LOGGER.info('Args preprocessing complete')
+
+    new_args = {
+        'args': rewritten_args,
+        'model_name': model_name,
+        'invest_version': __version__
+    }
```

Comment:
It may or may not be preferable, but at this point we could probably use `datastack.build_parameter_set`, to handle writing the json file.

Reply:
The code it would replace is below, not in the diff.

Reply:
That's a great idea!  It removes the relative path stuff to, since that's all handled within the parameter set code.  eb1fcc739